.pipelines/templates/build.yml

```diff
@@ -79,7 +83,7 @@ stages:
     - template: steps/full-checkout.yml
 
     - pwsh: |
-        ./build-package.ps1 -PackageName "${{ parameters.packageName }}" -StagedArtifactsPath "$(Build.ArtifactStagingDirectory)/package"
+        ./build-package.ps1 -PackageName "${{ parameters.packageName }}" -StagedArtifactsPath "$(Build.ArtifactStagingDirectory)/package" -TargetOS $(osName)
```

Comment:
Nit: I wonder if "platform" might be a better way to refer to what we are building for, now that WASM is in the mix?  For example, using `-TargetPlatform` here, instead?  We do have code that checks what OS it is building on, so those two concepts might be a little confusing to people down the road.

Reply:
Good idea, especially since WASM isn't really an OS?

To make this even more explicit, we could rename the old OS methods to indicate it's the "HostOS". That is, the OS that everything is building on, so we can more clearly differentiate between the Host OS/Platform and the Target OS/Platform. I'll do that :+1:

Reply:
Looks good!